Sources/CacheAPI-sqlite.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/**
+ * Simple Machines Forum (SMF)
+ *
+ * @package SMF
+ * @author Simple Machines http://www.simplemachines.org
+ * @copyright 2017 Simple Machines and individual contributors
+ * @license http://www.simplemachines.org/about/smf/license.php BSD
+ *
+ * @version 2.1 Beta 3
+ */
+
+if (!defined('SMF'))
+	die('Hacking attempt...');
+
+/**
+ * SQLite Cache API class
+ * @package cacheAPI
+ */
+class sqlite_cache extends cache_api
+{
+  /**
+   * @var string The path to the current $cachedir directory.
+   */
+  private $cachedir		= null;
+  private $cacheDB		= null;
+  private $cacheTime	= null;
+
+	public function __construct()
+	{
+		global $modSettings;
+
+		parent::__construct();
+
+		// Set our default cachedir.
+		$this->setCachedir();
+
+		$database = $this->cachedir.'/'.'SQLite3Cache.db3';
+
+		$this->cacheDB = new SQLite3($database);
+		$this->cacheDB->busyTimeout(1000);
+		if (filesize($database) == 0) {
+			$this->cacheDB->exec('CREATE TABLE cache (key text unique, value blob, ttl int);');
+			$this->cacheDB->exec('CREATE INDEX ttls ON cache(ttl);');
+		}
+		$this->cacheTime	= time();
+
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function isSupported($test = false)
+	{
+		$supported = class_exists("SQLite3") && is_writable($this->cachedir);
+
+		if ($test)
+			return $supported;
+
+		return parent::isSupported() && $supported;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function getData($key, $ttl = null)
+	{
+		$ttl		= time();
```

Comment:
We issue the ttl in the get as well as put as some cache methods need it.  But it can also be used to check to make sure that the cache is still valid.